sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -239,6 +240,7 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
         ignoreIfExists)
```

Comment:
shall we convert the table schema here, other than inside `createDataSourceTable`? The code looks clearer if we convert the table schema at the similar place for both Hive serde and data source tables.

Reply:
ah you are right